compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp

```diff
@@ -430,6 +435,12 @@ struct FusionOfTensorOpsPass
               return false;
             }
 
+            // Do not fuse if consumer is a contraction/matmul like op.
```

Comment:
Same here. These patterns propagate the reshapes down. So you would expect if there is a reshap followed by a contraction op, you want to propagate it down so that the dequantization can be fused with this. It seems like these two conditions are mixed up.

Reply:
Same as above, maybe @raikonenfnu can explain this.

Reply:
we have example for this as well on the test, I will brush up on this as well before our meet. :)